### PR TITLE
Rework Flaky Test Reporter

### PR DESCRIPTION
**Pull request by upodroid**
This PR does a couple of things:

- Reworks Flaky Test Reporter to use Application Default Credentials instead of hardcoded `option.WithCredentials` parameter. This bad practice needs to be cleaned up.
- Enables Workload Identity
- Pod now runs with a k8s service accout that assumes a Google Service Account
- Added image autobumper for it
- Added prow jobs to test this image.

/cc @kvmware @chizhg

### CODE REVIEW

File: tools/flaky-test-retryer/handler.go [hunk 1]
```diff
@@ -45,18 +45,18 @@ type HandlerClient struct {
...
+func NewHandlerClient(githubAccount string, dryrun bool) (*HandlerClient, error) {
 	ctx := context.Background()
-	if err := InitLogParser(serviceAccount); err != nil {
+	if err := InitLogParser(); err != nil {
 		log.Fatalf("Failed authenticating GCS: '%v'", err)
 	}
 	githubClient, err := NewGithubClient(githubAccount, dryrun)
 	if err != nil {
-		return nil, fmt.Errorf("Github client: %v", err)
+		return nil, fmt.Errorf("github client: %v", err)
```
Comment (chizhg) [inline tools/flaky-test-retryer/handler.go:55]: 👍

File: prow/jobs/custom/test-infra.yaml [hunk 2]
```diff
@@ -104,6 +104,40 @@ presubmits:
       - name: test-account
         secret:
           secretName: test-account
+  - name: pull-knative-test-infra-flaky-test-reporter-image-build
```
[the remark this answers is not in the document]
Comment (upodroid) [reply prow/jobs/custom/test-infra.yaml:107]: changed it to 4gb limit

File: prow/autobump-config/knative-prow-tests.yaml [hunk 3]
```diff
@@ -13,7 +13,7 @@ includedConfigPaths:
   - "prow/jobs_config"
 prefixes:
   - name: "Knative prow-tests image"
-    prefix: "gcr.io/knative-tests/test-infra/prow-tests"
+    prefix: "gcr.io/knative-tests/test-infra/flaky-test-reporter"
```
[the remark this answers is not in the document]
Comment (upodroid) [reply prow/autobump-config/knative-prow-tests.yaml]: it is typo, fixed

File: prow/jobs/custom/test-infra.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (chizhg) [reply prow/jobs/custom/test-infra.yaml:107]: Not sure if 4gb limit will work though, but let's see.

**Comment by upodroid** [comment]
Interesting, need to log a followup PR:
- forgot to enable DinD on the job
- The projects/knative-tests/subscriptions/flaky-test-retryer subscription is referring to a non existant topic.

**Pull request merged**